### ISSUE TITLE
Explicit action parameters

### ISSUE THREAD

**Issue by ZdenekM** [robofit/arcor2#309]
### Motivation

Currently, parameters of actions are stored in the `Resources` class and are referenced using `res.action_name`. This solution was introduced in order to allow:
- backward translation of code to its JSON representation,
- identify the concrete action during script execution.

However, it:

- makes script hard to read and understand,
- complicates possible usage of outputs of previous actions as parameters,
- complicates adjustment of parameters in code (need to edit project.json, leading to switching between Python/JSON).

The solution is to have explicit parameters generated during the build process and action name as a first parameter of actions:

```python
# examples of the current code
# robot_srv.get_input(res.action_name)
# actions.action_name()

# proposed solution
# action name parameter is only used by ARCOR2 and is not passed to the underlying REST call
robot_input_1_value = robot_srv.get_input("action_name", "aubo", "DI_1")
robot.srv.set_output("action_name_2", "aubo", "DO_3", robot_input_1_value)
```

Benefits: improved readability, ability to easily combine parameters given in the project with results of previous actions.

### Action state reporting

Currently, when the action is executed (during script execution), the following events are sent:

- `CurrentAction` (contains action_id, called from generated `Resources` class)
- `ActionState`
- ...now action is actually executed
- `ActionState` (containing results)

It would be probably a little bit cleaner to have `ActionStateBefore` (maybe containing parameters, depends on #366) and `ActionStateAfter` (containing results). Currently, `results` property is set only if `where == StateEnum.AFTER`.

With explicit parameters, this has to change. The ideal state would be:

- `ActionStateBefore` (containing action ID - if available / defined in project)
- ...now action is actually executed
- `ActionStateAfter` (containing results)

**Comment by ZdenekM** [comment]
For action points as parameters, we might extend the already generated `ActionPoints` class to provide properties also for joints and orientations. The class was so far used only when manually writing the script. It used to look like this:

```python
from arcor2.data.common import ActionPoint
from resources import Resources

class ActionPoints():

    def __init__(self, res: Resources):
        self._res = res

    @property
    def global_ap(self) -> ActionPoint:
        return self._res.project.action_point('ap_id')
```

Now it has to be modified to something like this:

```python
from arcor2.data.common import ActionPoint
from resources import Resources

class GlobalApOrientations():
    def __init__(self, res: Resources):
        self.default = res.project.pose("orientation_id")  # position of AP + orientation = Pose

class GlobalAp():
    def __init__(self, res: Resources):
        self.orientations = GlobalApOrientations(res)
        # self.joints = ...

class ActionPoints():

    def __init__(self, res: Resources):
        self.global_ap = GlobalAp(res)
```

And its usage in the main script would be following:

```python
from action_points import ActionPoints
...
aps = ActionPoints(res)
...
robot_srv.move("move_action_name", "eef", aps.global_ap.default)
```

Newly, the `Resources` class will automatically take care of transforming all relative poses into global ones before the execution of any actions (so far there was a `make_all_poses_absolute` method for that, intended to be called when manually writing the script). This transformation was up to now done in the `parameters` function which is called each time an action is called - that's not efficient, as we only work with global poses so it makes sense to transform them just once. Anyway, the `parameters` function will become obsolete with the explicit parameters.

**Comment by ZdenekM** [comment]
In contrast to the original design, I think it will be better to have an action name as the last, keyword-only, optional parameter.

**Comment by ZdenekM** [comment]
Implemented in #422.
